src/container.cpp

```diff
@@ -708,26 +708,26 @@ void Container::postRemoveNotification(Thing* thing, const Cylinder* newParent,
 	}
 }
 
-void Container::internalRemoveThing(Thing* thing)
+void Container::internalRemoveThing(std::shared_ptr<Thing> thing)
 {
-	auto cit = std::find(itemlist.begin(), itemlist.end(), thing);
-	if (cit == itemlist.end()) {
+	auto cit = std::find(itemList.begin(), itemList.end(), thing);
+	if (cit == itemList.end()) {
 		return;
 	}
-	itemlist.erase(cit);
+	itemList.erase(cit);
 }
 
-void Container::internalAddThing(Thing* thing) { internalAddThing(0, thing); }
+void Container::internalAddThing(std::shared_ptr<Thing> thing) { internalAddThing(0, thing); }
 
-void Container::internalAddThing(uint32_t, Thing* thing)
+void Container::internalAddThing(uint32_t, std::shared_ptr<Thing> thing)
 {
-	Item* item = thing->getItem();
+	auto item = thing->getItem();
```

Comment:
Could this be `const auto`
```suggestion
	const auto item = thing->getItem();
```